src/schemaType/BaseSchemaType.ts

```diff
@@ -64,7 +64,10 @@ abstract class BaseSchemaType {
 	public abstract set(record: MvRecord, value: unknown): MvRecord;
 
 	/** Validate value */
-	public abstract validate(value: unknown, document: Document): Promise<string[]>;
+	public abstract validate(
+		value: unknown,
+		document: Document,
+	): Promise<string[] | Map<string, string | string[]> | Map<string, string | string[]>[]>;
```

Comment:
Can this return type be simplified? I'd rather see this be `string[] | Map<string, string[]>`. There's a lot of potential combinations here and it complicates the analysis of the response.